base_layer/contacts/src/contacts_service/service.rs

```diff
@@ -620,7 +620,7 @@ where T: ContactsBackend + 'static
         &mut self,
         message: &Message,
     ) -> Result<(), ContactsServiceError> {
-        let address = &message.sender_address;
+        let address = &message.receiver_address;
```

Comment:
This doesn't seem correct to me.

The message is a received message, so we should still be the receiver address in this case. Meaning the confirmation needs to be sent to the `sender`